django_plastic_tickets/models.py

```diff
@@ -42,42 +58,90 @@ class ProductionMethod(DescribedModel):
 class MaterialType(DescribedModel):
     production_method = models.ForeignKey(ProductionMethod,
                                           on_delete=models.CASCADE)
+    density = models.FloatField(
+        help_text=gettext('Density in g/cm^3'),
+        blank=True, null=True)
 
     def __str__(self) -> str:
         return f'{self.name} ({self.production_method})'
 
 
-class MaterialColor(DescribedModel):
-    pass
+class Manufacturer(DescribedModel):
+    url = models.URLField(blank=True, null=True)
 
 
 class Material(models.Model):
```

Comment:
To my understanding of this, if Material was abstract class then it would not appear in the database. Thus, we could not reference it from MaterialStock. We can make Material abstract, but then we would have to add a field to MaterialStock (one to reference MaterialFilament and one to reference MaterialResin) which would not be very nice. There are also other approaches, but using Multi-table inheritance here is the best option I think.

Reply:
Ok agreeing with that